src/systems/joint_trajectory_controller/JointTrajectoryController.cc

```diff
@@ -0,0 +1,792 @@
+  this->target.acceleration = 0.0;
+  this->target.effort = 0.0;
+
+  this->pids.position = ignition::math::PID(
+      GetNthOrDefault(ParseVectorParam<double>(_sdf->Get<std::string>("position_p_gain")),
+                      _jointIndex, 0.0),
+      GetNthOrDefault(ParseVectorParam<double>(_sdf->Get<std::string>("position_i_gain")),
+                      _jointIndex, 0.0),
+      GetNthOrDefault(ParseVectorParam<double>(_sdf->Get<std::string>("position_d_gain")),
+                      _jointIndex, 0.0),
+      GetNthOrDefault(ParseVectorParam<double>(_sdf->Get<std::string>("position_i_max")),
+                      _jointIndex, -1.0),
+      GetNthOrDefault(ParseVectorParam<double>(_sdf->Get<std::string>("position_i_min")),
+                      _jointIndex, 0.0),
+      GetNthOrDefault(ParseVectorParam<double>(_sdf->Get<std::string>("position_cmd_min")),
+                      _jointIndex, -1.0),
+      GetNthOrDefault(ParseVectorParam<double>(_sdf->Get<std::string>("position_cmd_max")),
+                      _jointIndex, 0.0),
+      GetNthOrDefault(ParseVectorParam<double>(_sdf->Get<std::string>("position_cmd_offset")),
+                      _jointIndex, 0.0));
+
+  this->pids.velocity = ignition::math::PID(
+      GetNthOrDefault(ParseVectorParam<double>(_sdf->Get<std::string>("velocity_p_gain")),
+                      _jointIndex, 0.0),
+      GetNthOrDefault(ParseVectorParam<double>(_sdf->Get<std::string>("velocity_i_gain")),
+                      _jointIndex, 0.0),
+      GetNthOrDefault(ParseVectorParam<double>(_sdf->Get<std::string>("velocity_d_gain")),
+                      _jointIndex, 0.0),
+      GetNthOrDefault(ParseVectorParam<double>(_sdf->Get<std::string>("velocity_i_max")),
+                      _jointIndex, -1.0),
+      GetNthOrDefault(ParseVectorParam<double>(_sdf->Get<std::string>("velocity_i_min")),
+                      _jointIndex, 0.0),
+      GetNthOrDefault(ParseVectorParam<double>(_sdf->Get<std::string>("velocity_cmd_min")),
+                      _jointIndex, -1.0),
+      GetNthOrDefault(ParseVectorParam<double>(_sdf->Get<std::string>("velocity_cmd_max")),
+                      _jointIndex, 0.0),
+      GetNthOrDefault(ParseVectorParam<double>(_sdf->Get<std::string>("velocity_cmd_offset")),
+                      _jointIndex, 0.0));
+}
+
+bool ActuatedJoint::SetupComponents(ignition::gazebo::EntityComponentManager &_ecm) const
+{
+  const auto jointName = _ecm.Component<components::Name>(this->entity)->Data();
+
+  // Create JointPosition component if one does not exist
+  auto jointPositionComponent = _ecm.Component<components::JointPosition>(this->entity);
+  if (jointPositionComponent == nullptr)
+  {
+    _ecm.CreateComponent(this->entity, components::JointPosition());
+  }
+  if (jointPositionComponent == nullptr)
+  {
+    ignwarn << "[JointTrajectoryController] Cannot create JointPosition component for a joint ["
+            << jointName << "(Entity=" << this->entity << ")].\n";
```

Comment:
Do you know how you can end up in this situation? Does `CreateComponent` fail? And did you mean to assign a new value to `jointPositionComponent`?

The same goes for the other components below.

Reply:
> Do you know how you can end up in this situation? Does `CreateComponent` fail?

Oh yeah, it never fails. I guess I kept it there from my first attempt to create all components inside `Configure()`.

> And did you mean to assign a new value to `jointPositionComponent`?

Totally correct!

Simplified in https://github.com/ignitionrobotics/ign-gazebo/pull/473/commits/c462446ab88abb78c0edb47849e26af647a04e79